dashboard/app/controllers/java_builder_sessions_controller.rb

```diff
@@ -0,0 +1,25 @@
+require 'jwt'
+
+class JavaBuilderSessionsController < ApplicationController
+  authorize_resource class: false
+
+  PRIVATE_KEY = CDO.java_builder_private_key
+  PASSWORD = CDO.java_builder_key_password
+
+  # GET /javabuilder/access_token
+  def get_access_token
+    issued_at_time = Time.now.to_i
+    # expire token in 15 minutes
+    expiration_time = (Time.now + 15.minutes).to_i
+
+    payload = JWT.encode(
+      {
+        iat: issued_at_time,
+        exp: expiration_time
```

Comment:
should we include anything else here? One thing that comes to mind is user id, but I don't know if Java Builder will need it or not.

Reply:
JavaBuilder won't use the user id, but it will use the channel ID for the project. I don't think that needs to be included here, though, because the channel ID is public information.